### ISSUE TITLE
please help fine-tune the Edge of Chaos initializer

### ISSUE THREAD

**Issue by bionicles** [tensorflow/addons#556]
**Describe the feature and the current behavior/state.**
Criticality allows neural systems to maximize future possible states (because chaotic regimes have more diverse futures i guess) ... so here's an initializer which attempts to reach that

I can't find the paper right now but there was a good one which showed information can propagate infinite layers at the edge of chaos and just eyeballing their graphs it looked like the optimal value was around the value of Phi, so that's what i'm using here. 

How can we make this better?

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): see comments
- Is there already an implementation in another framework? (if so, where): idk
- Was it part of tf.contrib? (if so, where): idk

**Which API type would this fall under (layer, metric, optimizer, etc.)**
initializer

**Who will benefit with this feature?**
masters of chaos

**Any other info.**
```
# https://arxiv.org/pdf/1805.08266.pdf
# https://arxiv.org/pdf/1902.06853.pdf
import tensorflow as tf

# truncated_normal might be more stable
DISTRIBUTION = "normal"  

 # pareto's 20% rule to the 4th power ... reduce the power for a more stable initialization
PHI = 1.618 * (1. - (0.2 ** 4)) 

# not sure what to put here ... i wish there were a fan_both / fan_add for d_in + d_out
MODE = 'fan_avg'


class EdgeOfChaos(tf.keras.initializers.VarianceScaling):
    """The edge of chaos truncated normal initializer.
    It draws samples from a truncated normal distribution centered on 0
    with `stddev = sqrt(2 / (fan_in + fan_out))`
    where `fan_in` is the number of input units in the weight tensor
    and `fan_out` is the number of output units in the weight tensor.
    Args:
        kernel: boolean flag, if we're initializing a kernel, else do a bias
        fn_name: string name of the activation function
    References:
      [Glorot et al., 2010]
      http://jmlr.org/proceedings/papers/v9/glorot10a/glorot10a.pdf
      [Hayou et al., 2010]
      https://arxiv.org/pdf/1805.08266.pdf
    """

    def __init__(self, seed=None):
        super(EdgeOfChaos, self).__init__(
            distribution=DISTRIBUTION,
            scale=PHI,
            mode=MODE,
            seed=seed)

    def get_config(self):
        return {"seed": self.seed}
```

**Comment by bionicles** [comment]
ah, here's one paper with the phi thing, it's on page 2,  figure 1b, they say it's around stddev of weights 1.75 actually

**Comment by ghgit** [comment]
Perhaps this will help:
https://github.com/soufiane001/impact_of_init_and_activation/blob/master/eoc_curves.py
also:
https://github.com/ibm-research-tokyo/dybm/blob/master/src/pydybm/time_series/rnn_gaussian_dybm.py#L364

**Comment by bionicles** [comment]
Yes! Highly Active research area

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
